nilearn/input_data/base_masker.py

```diff
@@ -270,3 +270,61 @@ def _check_fitted(self):
             raise ValueError('It seems that %s has not been fitted. '
                              'You must call fit() before calling transform().'
                              % self.__class__.__name__)
+
+
+# XXX This function is converging toward filter_and_mask. They will merge
+# eventually.
+def filter_and_extract(imgs, extraction_function,
+                       smoothing_fwhm, t_r,
+                       standardize, detrend, low_pass, high_pass,
+                       confounds, memory, memory_level,
+                       target_fov=None, verbose=0):
+    """Extract representative time series using given function.
+
+    Parameters
+    ----------
+    imgs: images
+        Images to be masked
+
+    extraction_func: function
```

Comment:
So close and yet just slightly off the mark for the bonus points ;-). Still an extraction_func to go here and I'll merge this one.